File: src/main/java/com/waitring/waitring/repository/StoreRepositoryCustom.java
package com.waitring.waitring.repository;

import com.waitring.waitring.entity.Keyword;
import com.waitring.waitring.entity.Menu;
import com.waitring.waitring.entity.Store;

import java.util.List;

public interface StoreRepositoryCustom {
    List<Menu> getMenusByStore(Store store);
    List<Keyword> getKeywordsByStore(Store store);
    List<Store> getStoreListByWord(String word);
}

File: src/main/java/com/waitring/waitring/repository/StoreRepositoryCustom.kt
package com.waitring.waitring.repository

import com.waitring.waitring.entity.Keyword
import com.waitring.waitring.entity.Menu
import com.waitring.waitring.entity.Store

interface StoreRepositoryCustom {
    fun getMenusByStore(store: Store): List<Menu>
    fun getKeywordsByStore(store: Store): List<Keyword>
    fun getStoreListByWord(word: String): List<Store>
}